pkg/storagev2/migration/migration_test.go

```diff
@@ -0,0 +1,279 @@
+				},
+				1: func(s storage.Store) error {
+					return s.Put(objT3)
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "desc order version missing",
+			input: StepsMap{
+				4: func(s storage.Store) error {
+					return s.Put(objT1)
+				},
+				2: func(s storage.Store) error {
+					return s.Put(objT2)
+				},
+				1: func(s storage.Store) error {
+					return s.Put(objT3)
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if err := ValidateVersions(tt.input); (err != nil) != tt.wantErr {
+				t.Errorf("ValidateVersions() unexpected error: %v, wantErr : %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestMigrate(t *testing.T) {
+	t.Parallel()
+	objT1 := &obj1{Id: "aaa", SomeInt: 1}
+	objT2 := &obj1{Id: "bbb", SomeInt: 2}
+	objT3 := &obj1{Id: "ccc", SomeInt: 3}
+
+	t.Run("migration: 0 to 3", func(t *testing.T) {
+		t.Parallel()
+
+		steps := StepsMap{
+			1: func(s storage.Store) error {
+				return s.Put(objT1)
+			},
+			2: func(s storage.Store) error {
+				return s.Put(objT2)
+			},
+			3: func(s storage.Store) error {
+				return s.Put(objT3)
+			},
+		}
+
+		s := inmemstore.New()
+
+		if err := Migrate(s, steps); err != nil {
+			t.Errorf("Migrate() unexpected error: %v", err)
+		}
+
+		newVersion, err := GetVersion(s)
+		if err != nil {
+			t.Errorf("GetVersion() unexpected error: %v", err)
+		}
+		if newVersion != 3 {
+			t.Errorf("new version = %v must be 3", newVersion)
+		}
+
+		assertObjectExists(t, s, objT1, objT2, objT3)
+	})
+
+	t.Run("migration: 5 to 8", func(t *testing.T) {
+		t.Parallel()
+
+		steps := StepsMap{
+			8: func(s storage.Store) error {
+				return s.Put(objT1)
+			},
+			7: func(s storage.Store) error {
+				return s.Put(objT2)
+			},
+			6: func(s storage.Store) error {
+				return s.Put(objT3)
+			},
+		}
+
+		s := inmemstore.New()
+
+		err := SetVersion(s, 5)
+		if err != nil {
+			t.Errorf("SetVersion() unexpected error: %v", err)
+		}
+
+		if err := Migrate(s, steps); err != nil {
+			t.Errorf("Migrate() unexpected error: %v", err)
+		}
+
+		newVersion, err := GetVersion(s)
+		if err != nil {
+			t.Errorf("GetVersion() unexpected error: %v", err)
+		}
+		if newVersion != 8 {
+			t.Errorf("new version = %v must be 8", newVersion)
+		}
+
+		assertObjectExists(t, s, objT1, objT2, objT3)
+	})
+}
+func assertObjectExists(t *testing.T, s storage.Store, keys ...storage.Key) {
+	t.Helper()
+	for _, key := range keys {
+		if isExist, _ := s.Has(key); !isExist {
+			t.Errorf("key = %v doesn't exists", key)
+		}
+	}
+}
+
+func TestTagIDAddressItem_MarshalAndUnmarshal(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		test *storagetest.ItemMarshalAndUnmarshalTest
+	}{
+		{
+			name: "zero values",
+			test: &storagetest.ItemMarshalAndUnmarshalTest{
+				Item:    &StorageVersionItem{},
+				Factory: func() storage.Item { return new(StorageVersionItem) },
+			},
+		},
+		{
+			name: "min value",
```

Comment:
`1` is not a minimum value.
The test should cover the following cases:
- min value (already covered in "zero values")
- max value
- random value (within the min - max range)
- invalid size